File: 2540-minimum-common-value/2540-minimum-common-value.java
class Solution {
    public int getCommon(int[] nums1, int[] nums2) {
        HashSet<Integer> set = new HashSet<>();
        int len1 = nums1.length, len2 = nums2.length;
        
        for(int i = 0; i < len1; i++) {
            set.add(nums1[i]);
        }
        
        for (int i = 0; i < len2; i++) {
            if(set.contains(nums2[i])) {
                return nums2[i];
            }
        }
        
        return -1;
    }
}

File: 2540-minimum-common-value/2540-minimum-common-value.py
class Solution:
    def getCommon(self, nums1: List[int], nums2: List[int]) -> int:
        cur1 = 0
        cur2 = 0
        len1 = len(nums1)
        len2 = len(nums2)

        nums1.sort()
        nums2.sort()

        while cur1 < len1 and cur2 < len2:
            if nums1[cur1] < nums2[cur2]:
                cur1 += 1
            elif nums1[cur1] > nums2[cur2]:
                cur2 += 1
            else:
                break

        if cur1 == len1 or cur2 == len2:
            return -1

        return nums1[cur1]
